### DISCUSSION TITLE
Is Delete Account supported?

### DISCUSSION THREAD

**Discussion by vpyadus** [keycloakify/keycloakify#386] [Q&A] [answered]
Hello,
We need to allow a registered user to delete their account from the account console page. 

I've performed the following in Keycloak:
- Enabled the Delete Account action at Admin Console -> Authentication -> Required Actions tab.
- Added the delete-account role to the default user role. 

As a result, the Delete Account link is supposed to be displayed on the Account Console -> Personal Info page.
Unfortunately, it haven't appeared.

Please let me know if the Delete Account supported in Keycloakify, and if so, how can I make it visible?
Thanks in advance.

**Comment by garronej** [comment]
Hi @vpyadus,  

There is no blocker to implementing whatever Keycloak enable in a Keycloakify theme.  (Except [the wellcom page](https://github.com/keycloakify/keycloakify/assets/6702424/aa7b984e-783e-4e3c-9d78-d41e2cb90544)).  

That being sait maybe that the default page that we provide dosen't support this by default.  

1) Select the default keycloak theme. See if you see the button appear.
2) If you do, look at the kcContext for example by tryping kcContext in the chrome debugger and see what's in there and how you can use it to add the button you need.  Best is to use the  

You can use `npx download-builtin-keycloak-theme` to download the source of the default theme and see how they do it.  

Hope it helps

**Comment by vpyadus** [comment]
Thanks for the response!

I suppose that the default keycloak theme is `keycloak.v2`. If it's selected, the Delete Account link does appear on the user profile page.

Thanks for the recommendations, will try to use them and let you know if they don't work.

**Reply by garronej** [reply]
<del>I garenty 100% that it's doable, I just can tel if it's a oneliner or if it require more ui work</del>

**Reply by Saadkhh** [reply]
Hi @vpyadus, do you have a solution on making the delete account link appear using keycloakify?

**Reply by garronej** [reply, marked as answer]
Well despite the fact that I guarenteed 100% it's actually **not** possible at the moment.  🥇

**Reply by vpyadus** [reply]
Thanks for the question @Saadkhh.
No, I didn't manage to find an easy solution. We ended up using Keycloak Admin REST API for the purpose of deleting a user account.

**Comment by garronej** [comment]
Great thanks!  

If it's an easy fix it would be great to add it in the default page, let me know.

**Comment by garronej** [comment]
Hey @xgp!
This feature is not available in Account v1.
Since we are planning to maintain our own Java AccountResource, adding a feature to delete user accounts shouldn't be a big issue, right?

**Comment by xgp** [comment]
We will totally control the resource, so it should be trivial to customize.

**Reply by garronej** [reply]
We're very lucky to have you onboard to provide guidance on the Java side of things!
